### PR TITLE
Add information about Negative Response Caching

### LINKED ISSUE #743: Recommend disabling Negative DNS Response Caching on Windows
I ran into an problem today where a momentary flap during a service update caused one of my servers to no longer see it the service entry for 5 minutes without an `ipconfig /flushdns`. Windows respects the 0 TTL of a positive response, but if Consul returns a failure response (i.e. no available endpoints for a service definition) Windows will, by default, cache the failure for 900 seconds.

It would be worth it to recommend disabling negative DNS caching on Windows (see [KB 318803](https://support.microsoft.com/kb/318803/)). This applies to all versions of Windows __including and after__ XP despite the article being for XP.

I'd be happy to submit a PR to put it in the docs, maybe at http://www.consul.io/docs/guides/dns-cache.html but it seems like it should be a bit more obvious since this is nasty when it happens.

### PR DESCRIPTION
**Pull request by highlyunavailable**
Fixes #743

### CODE REVIEW

File: website/source/docs/guides/dns-cache.html.markdown [hunk 1]
```diff
@@ -44,6 +44,22 @@ meaning no stale results may be served. The default for
...
+mean that services appear "down" for longer than they are actually unavailable
+when using DNS for service discovery.
+
+One common example is that Windows will default to caching negative responses
+for 15 minutes. DNS forwarders may also cache negative responses, with the same
+effect. To avoid this problem, check the negative response cache defaults for
+your client operating system and any DNS forwarder on the path between the
+client and Consul and set the cache values appropriately. In many cases
+"appropriately" simply is turning negative response caching off to get the best
+recovery time to service when a service becomes available again.
```
Comment (ryanbreen) [inline website/source/docs/guides/dns-cache.html.markdown]: This looks really good!  I just suggest removing the "to service" in this last line here.

**Comment by highlyunavailable** [comment]
Thanks for your comment, I'm not sure how that slipped in there. Also what's your wrap length? It looks wider than mine - 80 perhaps? Mine is set at 72.

**Comment by ryanbreen** [comment]
Thanks!

**Comment by ryanbreen** [comment]
I tend not to wrap locally: the wrapping you were seeing in my comment was
whatever Github does by default.

On Mon, Mar 2, 2015 at 10:06 AM, highlyunavailable <notifications@github.com

**Pull request merged**